proxy/http/HttpSM.cc

```diff
@@ -3462,8 +3461,6 @@ HttpSM::tunnel_handler_for_partial_post(int event, void * /* data ATS_UNUSED */)
   tunnel.deallocate_buffers();
   tunnel.reset();
 
-  tunnel.allocate_redirect_postdata_producer_buffer();
-
```

Comment:
This line will cause the post buf truncated.

We should not deallocate `postbuf.ua_buffer_reader ` because it is the same as we doing `postbuf.ua_buffer_reader->consume`.